docs/up-to-date-check.md

```diff
@@ -195,6 +195,23 @@ The logs captured by that tool are usually adequate to diagnose build problems.
 
 ---
 
+## CopyToOutputDirectory Always vs. PreserveNewest
+
+When an item is to be copied to the output directly, there are two options:
+
+1. `CopyToOutputDirectory="Always"` always copies source over destination.
+2. `CopyToOutputDirectory="PreserveNewest"` copies the file is the timestamp of the source is newer than the destination.
+
+In reality, the use of `Always` is almost never what you actually want, yet it seems like a reasonable choice at first blush.
+
+If `Always` implies the file must be copied, even if it appears to be identical, then the presence of an `Always` item in the must cause the fast up-to-date check to immediately flag the project as out of date, leading to a build that can perform the copy.
+
+It is very easy for users to select the `Always` option and unwittingly break incremental build performance.
+
+In practice, there does not seem to be a valid use case for `Always`. The fast up-to-date check in VS therefore treats it as identical to `PreserveNewest` and only schedules a build if the timestamps or file sizes differ.
```

Comment:
Since we're comparing timestamps, this still covers the case where the file is edited by the application at run time, and the user wants it to be replaced with a "fresh" copy when building, correct?

Also, you may want to document this as the one valid case for `Always` that we are aware of.

Reply:
If the application modifies the file in the output directory at run time, then that output directory item will be newer.

Now that I think of it, I believe we can make this work correctly for Always too:

- For PreserveNewest, schedule a build if the source is different or newer than the destination
- For Always, schedule a build if the source is different or newer or older than the destination

I'll take another pass at this PR and get that fixed, and update the documentation as per your suggestions. Thanks.

Reply:
> I'll take another pass at this PR and get that fixed, and update the documentation as per your suggestions.

Done and done.